nuxeo-features/nuxeo-platform-audit/nuxeo-platform-audit-mongodb/src/main/java/org/nuxeo/mongodb/audit/pageprovider/MongoDBAuditPageProvider.java

```diff
@@ -118,7 +123,8 @@ protected void preprocessCommentsIfNeeded(List<LogEntry> entries) {
         List<LogEntry> entries = new ArrayList<>();
 
         // set total number of results
-        setResultsCount(auditCollection.count(filter));
+        MongoDatabase database = Framework.getService(MongoDBConnectionService.class).getDatabase(AUDIT_DATABASE_ID);
+        setResultsCount(countDocuments(database, auditCollection, filter));
```

Comment:
As we need to get `MongoDBConnectionService` to get the database, is there any cons to have an api on this service to count documents with correct MongoDB API? This could save some lines and static stuff in helpers.

Reply:
The origin of the static helper was based on the idea that we could rely on the collection's db name to discriminate databases. So in that context we could avoid making long calls to the Framework to call the service to get the database.

Now that this is not relevant anymore (since as you say, we need to get through this anyway) maybe we can update the `MongoDBConnectionService`

But then, wouldn't we then need to forward port this for people who migrate from 10.10 to 11.1 in the future ?

Reply:
I don't think we need to forward this as 11.x will be compatible with MongoDB starting from 4.0.

People migrating will have to upgrade their MongoDB instance(s) before upgrading Nuxeo.

Reply:
I'd rather not have public count APIs available in the service that would disappear in 11.1. Let's keep everything in the helper for now, even if it's slightly non-optimal.